ecosystem/typescript/sdk_v2/src/transactions/types/typeTag.ts

```diff
@@ -0,0 +1,488 @@
+    if (
+      this.value.module_name.identifier === "string" &&
+      this.value.name.identifier === "String" &&
+      this.value.address.toString() === AccountAddress.ONE.toString()
+    ) {
+      return true;
+    }
+    return false;
+  }
+}
+
+export class StructTag {
+  constructor(
+    public readonly address: AccountAddress,
+    public readonly module_name: Identifier,
+    public readonly name: Identifier,
+    public readonly type_args: Array<TypeTag>,
+  ) {}
+
+  /**
+   * Converts a string literal to a StructTag
+   * @param structTag String literal in format "AcountAddress::module_name::ResourceName",
+   *   e.g. "0x1::aptos_coin::AptosCoin"
+   * @returns
+   */
+  static fromString(structTag: string): StructTag {
+    // Use the TypeTagParser to parse the string literal into a TypeTagStruct
+    const typeTagStruct = new TypeTagParser(structTag).parseTypeTag() as TypeTagStruct;
+
+    // Convert and return as a StructTag
+    return new StructTag(
+      typeTagStruct.value.address,
+      typeTagStruct.value.module_name,
+      typeTagStruct.value.name,
+      typeTagStruct.value.type_args,
+    );
+  }
+
+  serialize(serializer: Serializer): void {
+    serializer.serialize(this.address);
+    serializer.serialize(this.module_name);
+    serializer.serialize(this.name);
+    serializer.serializeVector(this.type_args);
+  }
+
+  static deserialize(deserializer: Deserializer): StructTag {
+    const address = AccountAddress.deserialize(deserializer);
+    const moduleName = Identifier.deserialize(deserializer);
+    const name = Identifier.deserialize(deserializer);
+    const typeArgs = deserializer.deserializeVector(TypeTag);
+    return new StructTag(address, moduleName, name, typeArgs);
+  }
+}
+
+export const stringStructTag = new StructTag(
+  AccountAddress.fromHexInput({ input: "0x1" }),
+  new Identifier("string"),
+  new Identifier("String"),
+  [],
+);
+
+export function optionStructTag(typeArg: TypeTag): StructTag {
+  return new StructTag(
+    AccountAddress.fromHexInput({ input: "0x1" }),
+    new Identifier("option"),
+    new Identifier("Option"),
+    [typeArg],
+  );
+}
+
+export function objectStructTag(typeArg: TypeTag): StructTag {
+  return new StructTag(
+    AccountAddress.fromHexInput({ input: "0x1" }),
+    new Identifier("object"),
+    new Identifier("Object"),
+    [typeArg],
+  );
+}
+
+function bail(message: string) {
+  throw new TypeTagParserError(message);
+}
+
+function isWhiteSpace(c: string): boolean {
+  if (c.match(/\s/)) {
+    return true;
+  }
+  return false;
+}
+
+function isValidAlphabetic(c: string): boolean {
+  if (c.match(/[_A-Za-z0-9]/g)) {
+    return true;
+  }
+  return false;
+}
+
+// Generic format is T<digits> - for example T1, T2, T10
+function isGeneric(c: string): boolean {
+  if (c.match(/T\d+/g)) {
+    return true;
+  }
+  return false;
+}
+
+type TokenType = string;
+type TokenValue = string;
+type Token = [TokenType, TokenValue];
```

Comment:
is there a real reason to have these types? looks like TokenType only used in the Token type and TokenValue in only one more place other then the Token type?

Reply:
The more I see things like these, the more I realize they're all just aliased types purely for the sake of code readability.

Can remove it if you want, I think in this case it does increase the readability a bit, but if you think it's just more confusing than anything can just remove it

Reply:
but I will change it cuz it's just confusing